### PR TITLE
refactor(engine): merge two legacy interfaces together

### PR DESCRIPTION
**Pull request by pihme**
## Description

* `LegacyTypedStreamWriter` already extended `LegacyTypedCommandWriter`
* This PR replaces all references to `LegacyTypedCommandWriter` with `LegacyTypedStreamWriter` 
* It then merges the two interfaces and deleted `LegacyTypedCommandWriter`
* The net effect of this, is that we only need to replace one interface going forward

## Related issues

relates to #9724

## Definition of Done

Code changes:
* [X] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [X] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: engine/src/main/java/io/camunda/zeebe/engine/processing/streamprocessor/writers/LegacyTypedStreamWriter.java [hunk 1]
```diff
@@ -14,7 +14,13 @@
 
 /** Things that only a stream processor should write to the log stream (+ commands) */
 public interface LegacyTypedStreamWriter
-    extends LegacyTypedCommandWriter, TypedEventWriter, TypedRejectionWriter {
+    extends TypedCommandWriter, TypedEventWriter, TypedRejectionWriter {
```
Comment (Zelldon) [inline engine/src/main/java/io/camunda/zeebe/engine/processing/streamprocessor/writers/LegacyTypedStreamWriter.java:17]: :question: sorry I'm slightly confused now between the legacy and the without legacy interfaces :D what is the difference?

**Review by ChrisKujawa — APPROVED** [review]
Just so I get it right. The idea is that we then in the end just replace the Writer interface with the new Builder interface to create the result, correct? :) 

If this is the case makes sense to me.

File: engine/src/main/java/io/camunda/zeebe/engine/processing/streamprocessor/writers/LegacyTypedStreamWriter.java (hunk 1 above)
Comment (pihme) [reply engine/src/main/java/io/camunda/zeebe/engine/processing/streamprocessor/writers/LegacyTypedStreamWriter.java]: Legacy interface has `reset() + flush()`. The new ones don't

**Comment by pihme** [comment]
@Zelldon please have another look. I rebased everything and reduced the one legacy interface.

Now, the next step will be to replace usages of this one legacy interface with new tasks - scheduled via `ProcessingScheduleService`. This task will probably get an object similar to the response builder at runtime. I say similar, because I do want to define a reduced builder (i.e. result has only records, no side effects, no response)

Details still need to be figured out, but this is the general gist of it.

**Review by ChrisKujawa — APPROVED** [review]
Thanks :+1:

**Comment by pihme** [comment]
bors merge

**Pull request merged**